docs/framework/api-architecture.md

```diff
@@ -0,0 +1,175 @@
+# API architecture
+
+
+The term 'API' refers to that code stored in the top level api folder.
+It's important to note that only compliant code & usage will be
+supported. Usage documentation is in the [API
+v3](/api/index.md) page.
+
+## Standards {:#standards}
+
+-   The ONLY supported way to interact with any code in the API folder
+    from outside the api folder is by using the `civicrm_api()` wrapper
+-   Functionality delivered by the API is only supported if it is
+    1.  Advertised via the 'getfields function' OR (preferably AND)
+    2.  Verified by a test
+-   Where functions sit on the api but don't conform to the api
+    standards they may be supported by the creators but not by the
+    api team.
+
+## Individual functions
+
+-   All functions MUST have a success test as a minimum (failure tests
+    are only required for special features). API functions without tests
+    are not supported & are, in fact, not considered to be part of
+    the API.
+-   There are a number of (fairly new) helper functions you should use
+    in the test - please check [Writing a PHPUnit testcase
+    HOWTO](testing/setup.md)
+-   API functions should match the relevant BAO names & have the
+    functions 'get', 'create' and delete'
+-   All functions should receive $params as an array(not a reference)
+-   All functions should return either
+    -   an id indexed array of results via
+        the `civicrm_api3_create_success()` function
+    -   an appropriate message via throw new api_Exception
+-   (REVIEW this in light of feedback from bgm that it is not optimal
+    for translation) Do not use `ts()` in error messages (this will be
+    added in the exception function if you use api_Exception - but you
+    need to add a code)
+-   If a BAO object exists this must be passed to the `create_success` or
```

Comment:
I think we can ditch this one in favour of 'we are now officially confused whether create_success frees BAO objects' (just removing seems fine :-)